### PR TITLE
Fixed readInt

### PR DESCRIPTION
**Pull request by AndriyVel05**

### CODE REVIEW

**Review by YanOleks — COMMENTED** [review]
Не перевіряє вивід

**Review by smelovd — CHANGES_REQUESTED** [review]
fix output check

**Review by smelovd — CHANGES_REQUESTED** [review]
add param value to method check changes on dev

File: src/test/java/org/academy/codewars/ConsoleScannerTest.java [hunk 1]
```diff
@@ -20,26 +18,34 @@ public void beforeClass() {
...
+    public void testReadInt(String param, String input, int expectedValue, String expectedOutput) {
+        System.setIn(new ByteArrayInputStream(input.getBytes()));
+        ConsoleScanner consoleScanner = new ConsoleScanner(new Scanner(System.in));
+        var actualOutputStream = new ByteArrayOutputStream();
+        var originalOut = System.out;
+        System.setOut(new PrintStream(actualOutputStream));
+        int actualValue = consoleScanner.readInt(param);
+        System.setOut(originalOut);
+        assertEquals(actualValue, expectedValue);
+        String normalizedExpectedOutput = expectedOutput.replace("\r\n", "\n").replace("\r", "\n");
```
Comment (smelovd) [inline src/test/java/org/academy/codewars/ConsoleScannerTest.java:31]: remove

File: src/test/java/org/academy/codewars/ConsoleScannerTest.java [hunk 2]
```diff
@@ -20,26 +18,34 @@ public void beforeClass() {
...
+        String normalizedActualOutput = actualOutputStream.toString().replace("\r\n", "\n").replace("\r", "\n");
```
Comment (smelovd) [inline src/test/java/org/academy/codewars/ConsoleScannerTest.java:32]: remove

File: src/test/java/org/academy/codewars/dataproviders/ConsoleScannerDataProvider.java [hunk 3]
```diff
@@ -23,4 +23,19 @@ public Iterator<Object[]> dpTestReadInt() {
 
         return inputs.iterator();
     }
+
+    @DataProvider(name = "dp-testReadInt")
+    public Iterator<Object[]> testReadInt() {
+        List<Object[]> inputs = new ArrayList<>();
+        String param = "test int";
+        String expectedOutput1 = "Enter a " + param + " (int): ";
+        String expectedOutput2 = "Enter a " + param + " (int): Invalid data format. An integer is expected.\nEnter a " + param + " (int): ";
```
Comment (smelovd) [inline src/test/java/org/academy/codewars/dataproviders/ConsoleScannerDataProvider.java:32]: instead \n (only in expectedOutput) use ..." + System.lineSeparator() + "...,

**Review by smelovd — CHANGES_REQUESTED** [review]
remove all replaces

File: src/test/java/org/academy/codewars/ConsoleScannerTest.java [hunk 4]
```diff
@@ -20,32 +18,35 @@ public void beforeClass() {
...
+        System.setOut(originalOut);
+        assertEquals(actualValue, expectedValue);
+        assertEquals(actualOutputStream.toString(), expectedOutput);
     }
 
+
     @Test
     public void testReadIntInvalidData() {
-        String testData = "invalid\n25\n";
+        String testData = "invalid" + System.lineSeparator() + "25" + System.lineSeparator();
```
Comment (smelovd) [inline src/test/java/org/academy/codewars/ConsoleScannerTest.java:37]: move to dataprovider & delete method

File: src/test/java/org/academy/codewars/dataproviders/ConsoleScannerDataProvider.java [hunk 5]
```diff
@@ -8,22 +8,18 @@
...
-                "Element_1 => Enter a new string: Element_2 => Enter a new string: Element_3 => Enter a new string: ";
-
-        inputs.add(new Object[]{param, "3\nFirst arg \nSecond arg\nThird arg\n ", new String[]{"First arg", "Second arg", "Third arg"}, expectedOutput1});
-        inputs.add(new Object[]{param, "test\n-3\n3\nFirst arg \nSecond arg\nThird arg\n ", new String[]{"First arg", "Second arg", "Third arg"}, expectedOutput2});
-
-        return inputs.iterator();
-    }
+Fixed_readInt
+        @DataProvider(name = "dp-testReadInt")
+        public Iterator<Object[]> testReadInt() {
```
Comment (smelovd) [inline src/test/java/org/academy/codewars/dataproviders/ConsoleScannerDataProvider.java:13]: rename to dpTestReadInt

**Review by smelovd — CHANGES_REQUESTED** [review]

**Pull request closed without merging**